Snippet with same name as keyword does not expand

### Summary

<!-- Please insert a one line summary of the issue below -->

When selecting a completion entry which has 2 resolutions, 1 a keyword from the LSP, 1 from the extension snippets, only the keyword is selected even if the snippet is selected.

<!-- Be verbose: Include all steps necessary to reproduce from a clean Zed installation. -->
<!-- Code snippets are better than images, a repository link that reproduces the issue is ideal. -->

Steps to trigger the problem:
1. Install any Java JDK
2. Set JAVA_HOME environment variable to point to the JDK
3. Install "Java" Zed extension
4. Open a Java project (minimal reproducible project [ocpu/zed-bug-snippet-not-expanding](https://github.com/ocpu/zed-bug-snippet-not-expanding))
5. Write `class` (or `interface`, `record`)
6. Select the snippet entry from the completion list

Actual Behavior:
The word `class` is left alone in the file.
```java
class
```

Expected Behavior:
The snippet expands with the content of the snippet.
```java
package com.example.app;

public class Main {

    
}
```

Other non-conflicting names does work as shown in the reproduction video with `cls` user snippet.


<!--
  Is there anything additional necessary to reproduce this issue?
    - settings.json, keymap.json, .editorconfig etc?
    - Does it happen intermittently or only with specific projects / file types?
    - Have you found a workaround?

  Did you check your Zed.log to see if there is any relevant details there?
    - When including large items (videos, screenshots, logs, configs) please wrap with:

  <details><summary>See inside for XXXXYYY</summary>

  ```shell
  code
  ```

  </details>
-->

  <details><summary>Zed.log diff before and after trying to accept completion item</summary>

  ```log
  2025-02-27T17:35:12.063747191+01:00 [ERROR] Formatting failed: default prettier instance failed to format buffer: error during message '{"jsonrpc":"2.0","id":5,"method":"prettier/format","params":{"text":"..snip..","options":{"plugins":["/home/ocpu/.local/share/zed/prettier/node_modules/prettier-plugin-java/dist/index.js"],"parser":null,"filepath":"/home/ocpu/dev/zed-bug-snippet-not-expanding/src/main/java/com/example/app/Main.java","prettierOptions":{"printWidth":80,"tabWidth":4,"useTabs":false},"ignorePath":null}}}' handling: Error: Sad sad panda, parsing errors detected in line: NaN, column: NaN!
Expecting token of type --> Identifier <-- but found --> '' <--!
	->compilationUnit
	->ordinaryCompilationUnit
	->typeDeclaration
	->interfaceDeclaration
	->normalInterfaceDeclaration
	->typeIdentifier
2025-02-27T17:35:12.063890708+01:00 [ERROR] default prettier instance failed to format buffer

Caused by:
    error during message '{"jsonrpc":"2.0","id":5,"method":"prettier/format","params":{"text":"..snip..","options":{"plugins":["/home/ocpu/.local/share/zed/prettier/node_modules/prettier-plugin-java/dist/index.js"],"parser":null,"filepath":"/home/ocpu/dev/zed-bug-snippet-not-expanding/src/main/java/com/example/app/Main.java","prettierOptions":{"printWidth":80,"tabWidth":4,"useTabs":false},"ignorePath":null}}}' handling: Error: Sad sad panda, parsing errors detected in line: NaN, column: NaN!
    Expecting token of type --> Identifier <-- but found --> '' <--!
    	->compilationUnit
    	->ordinaryCompilationUnit
    	->typeDeclaration
    	->interfaceDeclaration
    	->normalInterfaceDeclaration
    	->typeIdentifier
  ```

  </details>
  <details><summary>Video of reproduction in Zed</summary>

https://github.com/user-attachments/assets/c97063cf-2033-431a-be5c-e6d59dcb731c

  </details>
  <details><summary>Video of expected behavior</summary>

https://github.com/user-attachments/assets/93efd4ea-8db6-49dd-a5cd-1b0308eb0d45

  </details>

### Zed Version and System Specs

Zed: v0.175.5 (Zed) 
OS: Linux Wayland ubuntu 24.04
Memory: 31 GiB
Architecture: x86_64
GPU: Intel(R) UHD Graphics (CML GT2) || Intel open-source Mesa driver || Mesa 24.2.8-1ubuntu1~24.04.1

--- a/crates/project/src/lsp_command.rs
+++ b/crates/project/src/lsp_command.rs
@@ -2092,11 +2092,16 @@ impl LspCommand for GetCompletions {
             completions.retain(|lsp_completion| {
                 let lsp_edit = lsp_completion.text_edit.clone().or_else(|| {
                     let default_text_edit = lsp_defaults.as_deref()?.edit_range.as_ref()?;
+                    let new_text = lsp_completion
+                        .insert_text
+                        .as_ref()
+                        .unwrap_or(&lsp_completion.label)
+                        .clone();
                     match default_text_edit {
                         CompletionListItemDefaultsEditRange::Range(range) => {
                             Some(lsp::CompletionTextEdit::Edit(lsp::TextEdit {
                                 range: *range,
-                                new_text: lsp_completion.label.clone(),
+                                new_text,
                             }))
                         }
                         CompletionListItemDefaultsEditRange::InsertAndReplace {
@@ -2104,7 +2109,7 @@ impl LspCommand for GetCompletions {
                             replace,
                         } => Some(lsp::CompletionTextEdit::InsertAndReplace(
                             lsp::InsertReplaceEdit {
-                                new_text: lsp_completion.label.clone(),
+                                new_text,
                                 insert: *insert,
                                 replace: *replace,
                             },
@@ -2167,6 +2172,7 @@ impl LspCommand for GetCompletions {
                                 .clone()
                         };
 
+                        // We already know text_edit is None here
                         let text = lsp_completion
                             .insert_text
                             .as_ref()

